Lib/typing.py

```diff
@@ -1922,10 +1926,9 @@ def _get_protocol_attrs(cls):
     """
     attrs = set()
     for base in cls.__mro__[:-1]:  # without object
-        if base.__name__ in ('Protocol', 'Generic'):
+        if base.__name__ in {'Protocol', 'Generic'}:
             continue
-        annotations = getattr(base, '__annotations__', {})
-        for attr in list(base.__dict__.keys()) + list(annotations.keys()):
+        for attr in (*base.__dict__, *base.__annotations__):
```

Comment:
How much does removing the `getattr` for `__annotations__` help? I'm not sure it's safe. E.g. classes from extension modules will usually not have `__annotations__`, but maybe there aren't any valid ways such a class could end up here.

Reply:
I haven't measured the impact in isolation, but I'm _pretty_ confident it's safe following the changes Larry made to `__annotations__` in 3.10+. From [the docs he wrote](https://docs.python.org/3/howto/annotations.html#accessing-the-annotations-dict-of-an-object-in-python-3-10-and-newer):

> If for some reason [inspect.get_annotations()](https://docs.python.org/3/library/inspect.html#inspect.get_annotations) isn’t viable for your use case, you may access the `__annotations__` data member manually. Best practice for this changed in Python 3.10 as well: as of Python 3.10, `o.__annotations__` is guaranteed to always work on Python functions, classes, and modules. If you’re certain the object you’re examining is one of these three specific objects, you may simply use `o.__annotations__` to get at the object’s annotations dict.

`__mro__` is a read-only attribute, so if a non-class ends up here, something seriously weird is going on, that imo we probably shouldn't have to worry about in typing.py

Reply:
Hmm, haven't kept up to date with the exact changes to `__annotations__`, but the word Python in "Python functions, classes and modules" may preclude extension modules:
```
λ python            
Python 3.10.9 (main, Dec 31 2022, 16:26:27) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import xxhash
>>> xxhash.xxh32.__annotations__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'xxhash.xxh32' has no attribute '__annotations__'
```

Reply:
Hum. Okay, I'll revert the change to directly access `__annotations__`. Thanks for the catch!

Imo the wording for the annotations docs could maybe be improved there, that's a bit subtle for my tastes. (Or maybe it's actually a bug in `xxhash`. Idk.)

Reply:
I think this is safe, but only because `Protocol.__init_subclass__` restricts the MRO of a `Protocol` class such that I don't think anything can appear in it which doesn't inherit from something that has `__annotations__`.

But that's a pretty subtle thing to implicitly rely on here, so it seems probably safest to keep the getattr.

Reply:
```suggestion
        annotations = getattr(base, '__annotations__', {})
        for attr in (*base.__dict__, *annotations):
```